.github/workflows/build-pr.yml

```diff
@@ -0,0 +1,38 @@
+name: Build PR
```

Comment:
👍🏿 for splitting check from the site preview deploy

Reply:
I's almost the same as the job to deploy the site. Maybe we should merge the common steps in another workflow later.